GameServerLib/GameObjects/AttackableUnits/AttackableUnit.cs

```diff
@@ -128,6 +128,10 @@ public virtual bool IsInDistress()
         public virtual void TakeDamage(IAttackableUnit attacker, float damage, DamageType type, DamageSource source,
             DamageText damageText)
         {
+            if (damage < 0f) {
+                Logger.Error("Damage must be non-negative");
```

Comment:
The log entry is a bit excessive don't you think? There are ways to deal negative damage in vanilla League of Legends as well, that shouldn't be an issue.

Reply:
I guess that this would be a fault from the spell/item/etc code.

Reply:
Also now that I think about it, this probably should be implemented by setting the damage to 0 rather than returning from the function. There might be cases where we need to track how many instances of damage taking events have passed, so 0 damage should follow through the same logic as 100 damage.

Reply:
done